### PR TITLE
Make demo app support AAD (similar to how test app already supports it)

### PR DESCRIPTION
**Pull request by micarls**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Test app already supports AAD. This makes demo app support it as well.

## Related PRs or issues

https://github.com/microsoft/appcenter-sdk-dotnet/pull/1112

## Misc

Tested the IOS and Android app secrets are working (via analytics)

### CODE REVIEW

**Comment by guperrot** [comment]
Was it tested with at least analytics?

**Comment by micarls** [comment]
I haven't tested with analytics yet. If that is the min bar, I will test with that.

**Comment by guperrot** [comment]
That change is only to test app secret switching, so any way to verify it uses the right appSecret after restart is good, crashes works too.

**Comment by micarls** [comment]
Makes sense, thanks.

**Comment by micarls** [comment]
Yes, I can confirm both IOS and Android telemetry points are getting hit. I consider this ready to merge now.

File: Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/App.xaml.cs [hunk 1]
```diff
@@ -25,10 +26,21 @@ public partial class App
...
+            { XamarinDevice.Android, "987b5941-4fac-4968-933e-98a7ff29237c" },
+            { XamarinDevice.iOS, "fe2bf05d-f4f9-48a6-83d9-ea8033fbb644" }
+        };
+
+        // App Center AAD secrets
+        static readonly IReadOnlyDictionary<string, string> AADAuthAppSecrets = new Dictionary<string, string>
+        {
+            { XamarinDevice.UWP, "5bce20c8-f00b-49ca-8580-7a49d5705d4c" }, // same for all devices
+            { XamarinDevice.Android, "739eacee-42de-454c-b0d7-c093e765e009" },
+            { XamarinDevice.iOS, "e9f015c6-6c2b-4410-8053-70eaa52d90e0" }
```
Comment (guperrot) [inline Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/App.xaml.cs]: This is missing plist file modifications for distribute and auth

Comment (micarls) [reply Apps/Contoso.Forms.Demo/Contoso.Forms.Demo/App.xaml.cs]: Good point, I've added it for both puppet (which was somehow missed) and demo. Still working on testing this new change.

**Pull request merged**